doc/Hammer-Use/Hooks.rst

```diff
@@ -1,31 +1,68 @@
+.. _hooks:
+
 Extending Hammer with Hooks
 =======================================
 
 It is unlikely that using the default Hammer APIs alone will produce DRC- and LVS-clean designs with good QoR in advanced technology nodes if the design is sufficiently complex.
 To solve that, Hammer is extensible using *hooks*.
 These hooks also afford power users additional flexibility to experiment with CAD tool commands to tweak aspects of their designs.
+The hook framework is inherently designed to enable reusability, because successful hook methods that solve a technology-specific problem may be upstreamed into the technology plugin for future designs.
 
 Hook Methods
 ------------
 
-Hooks are fundamentally Python methods that extend a given tool's set of available steps and can inject additional TCL commands into the flow. Hook methods need to take in an instance of a particular ``HammerTool``, which provides them with the full set of Hammer IR available to the tool.
+Hooks are fundamentally Python methods that extend a given tool's set of available steps and can inject additional TCL commands into the flow. 
+Hook methods need to take in an instance of a particular ``HammerTool``, which provides them with the full set of Hammer IR available to the tool.
+Hooks (depending on how they are included, see below) get turned into step objects that can be targeted with ``--from/after/to/until_step`` and other hooks.
 
-Hooks can live in a Python file inside the design root so that it is available to the class that needs to extend the default ``CLIDriver``. An example of some skeletons of hooks are found in `Chipyard <https://github.com/ucb-bar/chipyard/blob/master/vlsi/example-vlsi>`__.
+Hooks can live in a Python file inside the design root so that it is available to the class that needs to extend the default ``CLIDriver``.
+An example of some skeletons of hooks are found in `Chipyard <https://github.com/ucb-bar/chipyard/blob/master/vlsi/example-vlsi>`__.
```

Comment:
We can also link to some examples of unit tests which use hooks (which we hopefully have...)